### PR TITLE
[cxxmodules] Don't declare strings from rootmap for modules

### PR DESCRIPTION
**Pull request by yamaguchi1024**
We don't need any input_line declration from rootmap for startup time.
However it may cause a failure on tests when they have custom
dictionaries.

Gives 5MB impr for modules, but still Testing!

### CODE REVIEW

File: core/metacling/src/TCling.cxx [hunk 1]
```diff
@@ -5354,12 +5354,15 @@ Int_t TCling::LoadLibraryMap(const char* rootmapfile)
 
    // Process the forward declarations collected
    cling::Transaction* T = nullptr;
-   auto compRes= fInterpreter->declare(uniqueString.Data(), &T);
-   assert(cling::Interpreter::kSuccess == compRes && "A declaration in a rootmap could not be compiled");
 
-   if (compRes!=cling::Interpreter::kSuccess){
-      Warning("LoadLibraryMap",
+   if (!fCxxModulesEnabled) {
```
Comment (pcanal) [inline core/metacling/src/TCling.cxx]: Isn't the correct test: 'if module-is-enabled && this-library-has-a-pcm' ?

Comment (vgvassilev) [reply core/metacling/src/TCling.cxx]: That's the challenge. We do not have that information at that point. Worse, even we have a pcm, the loading of the pcm can fail.

Comment (pcanal) [reply core/metacling/src/TCling.cxx]: Well, it seems it is a challenge we must tackled.  *If* I am reading this correctly, this PR means that autoloading is disabled for library/dictionary that do not use pcm but module support is enabled (i.e. for all intent and purposes that would mean that we do **not** support the mixed case)

Comment (vgvassilev) [reply core/metacling/src/TCling.cxx]: It seems we can check here if we have a pcm with the expected name on disk. We should change TCling::RegisterModule such that if we fail to load the nominated pcm, we should alternatively load the corresponding rootmap. We are somewhat safe with that solution as the current implementation preloads all available pcm files and virtually LoadLibraryMap happens at the same time as LoadModule.

Comment (yamaguchi1024) [reply core/metacling/src/TCling.cxx]: requiresRootMap(rootmapfile, fInterpreter) is essentially doing the same thing as 'if module-is-enabled && this-library-has-a-pcm', but please give me some time to try those out ( do not review yet label ;) )

Comment (vgvassilev) [reply core/metacling/src/TCling.cxx]: Now we seem to have is interface as per `HasPCMForLibrary`. We should use it here.

**Comment by vgvassilev** [comment]
Superseded by PR #3043.

**Pull request closed without merging**